core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1029,6 +1010,17 @@ public GHResponse route(GHRequest request) {
         Lock readLock = readWriteLock.readLock();
         readLock.lock();
         try {
+            if (!request.getVehicle().isEmpty())
+                throw new IllegalArgumentException("GHRequest may no longer contain a vehicle, use the profile parameter instead, see #todonow");
+            if (!request.getWeighting().isEmpty())
+                throw new IllegalArgumentException("GHRequest may no longer contain a weighting, use the profile parameter instead, see #todonow");
+            if (!request.getHints().get(Routing.TURN_COSTS, "").isEmpty())
+                throw new IllegalArgumentException("GHRequest may no longer contain the turn_costs=true/false parameter, use the profile parameter instead, see #todonow");
+            // todonow: maybe still allow something like running a (non CH) profile edge-based or not (if no turn costs or something)?, also see traversal mode below
+            if (!request.getHints().get(Routing.EDGE_BASED, "").isEmpty())
+                throw new IllegalArgumentException("GHRequest may no longer contain the edge_based=true/false parameter, use the profile parameter instead, see #todonow");
+            // todonow: do not allow things like short_fastest.distance_factor or u_turn_costs unless CH is disabled and only under certain conditions for LM
```

Comment:
This is about the 'merging' feature where some properties are set right in the profile (like distance_factor), but they might be overwritten per request. The problem is that this does not work with CH (and only some special cases work for LM), so I am a bit unsure where to put these checks. They cannot really go into `createWeighting` (where the merging will happen?!), because in this method we do not 'know' whether the weighting is created for CH or not...

Reply:
for now we are just blindly merging the hints, later we will add these kind of checks here to be more strict